### PR TITLE
AppleHealthSleep - new types for Oura

### PR DESCRIPTION
**Pull request by brandoncharnesky**
## Overview

Addresses (https://github.com/CareEvolution/MyDataHelpsViewBuilder/issues/58)

The three new Oura ring sleep types (AsleepCore, AsleepDeep, AsleepREM) are now rolled up as as "Asleep" time.  There are three additional daily-data-providers added as well, for future use of breaking those sleep types out separately.  

AppleSleepData1
AppleSleepData2

## Security

REMINDER: All file contents are public.

- [x] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc.
  - [x] Please verify that you double checked that .storybook/preview.js does not contain your participant access key details. 
  - [x] There are no temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
- [x] These changes do not introduce any security risks, or any such risks have been properly mitigated.

Describe briefly what security risks you considered, why they don't apply, or how they've been mitigated.

### Testing
- [ ] MyDataHelps iOS app
- [ ] MyDataHelps Android app
- [x] [MyDataHelps website](mydatahelps.org)

### Documentation
- [ ] I have added relevant Storybook updates to this PR as well.
- [ ] If this feature requires a developer doc update, tag @CareEvolution/api-docs.

Consider "Squash and merge" as needed to keep the commit history reasonable on `main`.

## Reviewers

Assign to the appropriate reviewer(s). Minimally, a second set of eyes is needed ensure no non-public information is published. Consider also including:
- Subject-matter experts
- Style/editing reviewers
- Others requested by the content owner

### CODE REVIEW

File: src/helpers/daily-data-providers/apple-health-sleep.ts [hunk 1]
```diff
@@ -116,4 +123,20 @@ export function asleepTime(startDate: Date, endDate: Date) {
...
+
+export function asleepRemTime(startDate: Date, endDate: Date) {
+	return coreSleep(startDate, endDate, "AsleepREM");
+}
+
+export function asleepDeepTime(startDate: Date, endDate: Date) {
+	return coreSleep(startDate, endDate, "AsleepDeep");
+}
+
+export function asleepAnyTime(startDate: Date, endDate: Date) {
```
Comment (chrisnowak) [inline src/helpers/daily-data-providers/apple-health-sleep.ts:140]: Let's remove this one and instead we should update "asleepTime" function above to include ```["Asleep", "AsleepCore", "AsleepREM", "AsleepDeep" ]

File: src/components/container/DeviceDataMonthCharts/DeviceDataMonthCharts.tsx [hunk 2]
```diff
@@ -114,6 +114,17 @@ export default function (props: DeviceDataMonthChartsProps) {
 					month={month}
 					year={year}
 					syncId="DeviceDataCharts" />
+				<DeviceDataMonthChart onDataDetected={() => onDataDetected()}
```
Comment (chrisnowak) [inline src/components/container/DeviceDataMonthCharts/DeviceDataMonthCharts.tsx:117]: Can probably remove this chart from this standard viz, we aren't really using the DeviceDataMonthChart in view builder etc.

File: src/helpers/daily-data-providers/apple-health-sleep.ts [hunk 3]
```diff
@@ -110,10 +117,26 @@ function coreSleep(startDate: Date, endDate: Date, value: "Asleep" | "InBed") {
...
+
+export function asleepCoreTime(startDate: Date, endDate: Date) {
+	return coreSleep(startDate, endDate, "AsleepCore");
+}
+
+export function asleepRemTime(startDate: Date, endDate: Date) {
+	return coreSleep(startDate, endDate, "AsleepREM");
+}
+
+export function asleepDeepTime(startDate: Date, endDate: Date) {
```
Comment (chrisnowak) [inline src/helpers/daily-data-providers/apple-health-sleep.ts:136]: We should register these 3 new apple health sleep types in the daily data types:
https://github.com/CareEvolution/MyDataHelpsUI/blob/main/src/helpers/query-daily-data.ts#L119

https://github.com/CareEvolution/MyDataHelpsUI/blob/main/src/helpers/query-daily-data.ts#L181

(I would not do "asleep only"; don't think that is useful - just add core, rem, and deep)

File: src/helpers/daily-data-providers/apple-health-sleep.ts [hunk 4]
```diff
@@ -73,7 +73,14 @@ function calculateSleepTime(ranges: DateRange[]) {
...
 
-function coreSleep(startDate: Date, endDate: Date, value: "Asleep" | "InBed") {
+type SleepType = "Asleep" | "InBed" | "AsleepCore" | "AsleepREM" | "AsleepDeep";
+
+function coreSleep(startDate: Date, endDate: Date, value: SleepType) {
+	var val: SleepType[] = [ value ];
+	return coreSleepMultiValue(startDate, endDate, val);
+}
+
+function coreSleepMultiValue(startDate: Date, endDate: Date, value: SleepType[]) {
```
Comment (chrisnowak) [inline src/helpers/daily-data-providers/apple-health-sleep.ts:83]: I'd rename value > values

File: src/helpers/daily-data-providers/apple-health-sleep.ts [hunk 5]
```diff
@@ -85,7 +92,7 @@ function coreSleep(startDate: Date, endDate: Date, value: "Asleep" | "InBed") {
 
 		ddp.forEach((d) => {
 			if (!d.observationDate || !d.startDate) { return; }
-			if (d.value != value) { return; }
+			if (!value.find(x => x /* as string */ == d.value) ) { return; }
```
Comment (chrisnowak) [inline src/helpers/daily-data-providers/apple-health-sleep.ts:95]: remove comment (it's unnecessary right?)

Comment (brandoncharnesky) [reply src/helpers/daily-data-providers/apple-health-sleep.ts:95]: Thought I might have some issues with typescript, but it is not complaining

**Comment by chrisnowak** [comment]
The core change here looks good; I would just remove the DeviceDataMonthChart changes and make sure you register the new daily data types following the patterns in query-daily-data.ts

**Review by chrisnowak — APPROVED** [review]

**Pull request merged**